darts/utils/data/tabularization.py

```diff
@@ -165,144 +165,1181 @@ def _create_lagged_data(
+    past_covariates: Optional[TimeSeries] = None,
+    future_covariates: Optional[TimeSeries] = None,
+    lags: Optional[Sequence[int]] = None,
+    lags_past_covariates: Optional[Sequence[int]] = None,
+    lags_future_covariates: Optional[Sequence[int]] = None,
+    output_chunk_length: int = 1,
+    max_samples_per_ts: Optional[int] = None,
+    multi_models: bool = True,
+    check_inputs: bool = True,
+    use_moving_windows: bool = True,
+    is_training: bool = True,
+) -> Tuple[np.ndarray, np.ndarray, pd.Index]:
+
+    """
+    Creates the features array `X` and labels array `y` to train a lagged-variables regression model (e.g. an
+    `sklearn` model) when `is_training = True`; alternatively, creates the features array `X` to produce a series
+    of prediction from an already-trained regression model when `is_training = False`. In both cases, the time index
+    values of each generated observation is also returned.
+
+    Notes
+    -----
+    Instead of calling `create_lagged_data` directly, it is instead recommended that:
+        - `create_lagged_training_data` be called if one wishes to create the `X` and `y` arrays
+        to train a regression model.
+        - `create_lagged_prediction_data` be called if one wishes to create the `X` array required
+        to generate a prediction from an already-trained regression model.
+    This is because even though both of these functions are merely wrappers around `create_lagged_data`, their
+    call signatures are more easily interpreted than `create_lagged_data`. For example,
+    `create_lagged_prediction_data` does not accept `output_chunk_length` nor `multi_models` as inputs, since
+    these inputs are not used when constructing prediction data. Similarly, `create_lagged_prediction_data`
+    returns only `X` and `times` as outputs, as opposed to returning `y` as `None` along with `X` and `times`.
+
+    The `X` array is constructed from the lagged values of up to three separate timeseries:
+        1. The `target_series`, which contains the values we're trying to predict. A regression model that
+        uses previous values
+        of the target its predicting is referred to as *auto-regressive*; please refer to [1]_ for further
+        details
+        about auto-regressive timeseries models.
+        2. The past covariates series, which contains values that are *not* known into the future. Unlike
+        the target series, however,
+        past covariates are *not* to be predicted by the regression model.
+        3. The future covariates (AKA 'exogenous' covariates) series, which contains values that are known
+        into the future, even beyond
+        the data in `target_series` and `past_covariates`.
+    See [2]_ for a more detailed discussion about target, past, and future covariates. Conversely, `y` is
+    comprised only of the lagged values of `target_series`.
+
+    The shape of `X` is:
+        `X.shape = (n_observations, n_lagged_features, n_samples)`,
```

Comment:
That's right - there's no fundamental reason to keep the samples axis; I just thought there would be no real harm in allowing `create_lagged_data` to be vectorised over the samples axis. 

A potentially cool application for this would be to allow users to perform Monte Carlo-style uncertainty quantification with their models. For example, although a user may start out with a single deterministic timeseries, but they can perturb this series by random noise to generate a distribution of 'pseudo-data'. They could then fit their model to each one of these perturbed series and see the resulting distribution in model parameters as a way to quantify their uncertainty in those parameters.

The results of this Monte Carlo-style analysis for 'vanilla' linear regression models wouldn't exactly be surprising (since the resulting distribution is guaranteed to be normal - see Section 13.1 of [*All of Statistics* by Larry Wasserman](https://egrcc.github.io/docs/math/all-of-statistics.pdf)), but the distribution isn't necessarily normal for regularised linear regression (I think) nor for non-linear regression models. Thoughts?

Reply:
OK, nice, I like this idea, we can keep it :)
In fact, we are doing something similar (although not exactly the same thing) in `TorchForecastingModel`s, where if an input series is stochastic, we select one of its sample uniformly at random, which also allows similar Monte Carlo sampling capabilities.
Two small things maybe:
* It'd be nice to test this if possible (if you have already a test for this ignore this comment - I simply missed it :)
* We could maybe add a small note in the docstring of `RegressionModel` to describe what happens if the input series are themselves stochastic.